### ISSUE TITLE
require.resolve causes invalid package names

### ISSUE THREAD

**Issue by magnusottosson** [depcheck/depcheck#859]
### Bug Description

Im extending the [xo-typescript](https://github.com/xojs/eslint-config-xo-typescript/blob/main/index.js) eslint config in my `eslintrc.js` file. In this config it has the following line: `parser: require.resolve('@typescript-eslint/parser')`. When I run depcheck I get a missing dependency that reports like:

```
Missing dependencies
* /Users/username/my-path-to-my-module/node_modules/@typescript-eslint/parser/dist/index.js: ./.eslintrc.js
```

If I manually change the package in `node_modules` to `'@typescript-eslint/parser'` it doesnt report the dep as missing any more.

Anyway to handle this (since I cant modify the package)? 

### Versions

- `node -v`: v20.9.0
- `npm -v`: 9.8.1
- `depcheck --version`: 1.4.7

### Extra info

The affecting info from the --json output

```
"missing": {
    "/Users/username/some-path/my-module/node_modules/@typescript-eslint/parser/dist/index.js": [
      "/Users/username/some-path/my-module/.eslintrc.js"
    ]
  },
```
